Path/path.h

```diff
@@ -0,0 +1,51 @@
+#ifndef PATROL_PATH_H
+#define PATROL_PATH_H
+
+#include "../dependencies/glm/glm.hpp"
+#include "../r_common.h"
+#include <string>
+#include <vector>
+
+struct Waypoint {
+    glm::vec3 position;
+    int id;
+    int target;
+};
+
+class PatrolPath {
+  private:
+    std::vector<Waypoint> m_Points;
+    float m_Radius;
+    std::string m_name;
+    bool m_IsClosed;
+    int m_CurrentWaypointIndex;
```

Comment:
Not being initialized. Is this on purpose? Caused the program to crash because the Index was set to some random large negative number.